hvplot/cli.py

```diff
@@ -0,0 +1,179 @@
+https://hvplot.holoviz.org/user_guide/Customization.html
+
+Examples:
+    hvplot iris.csv
+    hvplot air.nc x=lon y=lat groupby=time,level geo=true
+    hvplot detrend.nino34.ascii.txt x=YR y=ANOM -r pandas.read_csv -rk delimiter='\s+'
+"""
+
+_MODULES_KWARGS = {
+    "xarray": {
+        "extensions": [".nc", ".hdf5", ".h5", ".zarr", ".grib"],
+        "default_reader": "open_dataset",
+        "extras": {
+            ".zarr": {"kwargs": {"engine": "zarr"}},
+            ".grib": {"kwargs": {"engine": "cfgrib"}},
+        },
+    },
+    "pandas": {
+        "extensions": [".csv", ".json", ".html", ".xls", ".xlsx", ".parquet"],
+        "default_reader": "read_csv",
+        "extras": {
+            ".json": {"reader": "read_json"},
+            ".html": {"reader": "read_html"},
+            ".xls": {"reader": "read_excel"},
+            ".xlsx": {"reader": "read_excel"},
+            ".parquet": {"reader": "read_parquet"},
+        },
+    },
+    "geopandas": {
+        "extensions": [".shp", ".geojson", ".gml"],
+        "default_reader": "read_file",
+    },
+}
+_EXTENSION_READERS = {}
+
+
+def import_and_register_module(module_name, extensions, default_reader, extras=None):
+    try:
+        module = __import__(module_name)
+        if module_name == "geopandas":
+            module_name = "pandas"  # Ensure hvplot.pandas is imported for geopandas
+        __import__("hvplot." + module_name)
+
+        for ext in extensions:
+            reader_kwargs = {"reader": default_reader}
+            if extras and ext in extras:
+                reader_kwargs.update(extras[ext])
+
+            if hasattr(module, reader_kwargs["reader"]):
+                _EXTENSION_READERS[ext] = (
+                    getattr(module, reader_kwargs["reader"]),
+                    reader_kwargs.get("kwargs", {}),
+                )
+    except ImportError:
+        pass
+
+
+def setup_extension_readers(extension):
+    for module_name, module_kwargs in _MODULES_KWARGS.items():
+        if extension in module_kwargs["extensions"]:
+            import_and_register_module(module_name, **module_kwargs)
+            break
+
+
+def parse_inputs(inputs):
+    kwargs = {}
+    if not inputs:
+        return kwargs
+
+    if isinstance(inputs, str):
+        inputs = [inputs]
+
+    for input_ in inputs:
+        key, value = input_.split("=")
+        if "," in value:
+            value = [v for v in value.strip("[]").strip("()").split(",") if v]
+        elif value.lower() in ("true", "false"):
+            value = value == "true"
+        kwargs[key] = value
+    return kwargs
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(
+        description=HELP_DESCRIPTION, formatter_class=argparse.RawTextHelpFormatter
+    )
+    parser.add_argument("file_path", type=str, help="Path to the data file")
+    parser.add_argument(
+        "hvplot_kwargs", nargs="*", help="HoloViews options in 'key=value' format"
```

Comment:
```suggestion
        "hvplot_kwargs", nargs="*", help="hvPlot arguments in 'key=value' format"
```

(Is that what you meant, to distinguish this from `--opts`?)